### DISCUSSION TITLE
Support xCode 13's test repetitions feature:"Run _tests Repeatedly"

### DISCUSSION THREAD

**Discussion by D3icidal** [fastlane/fastlane#19439] [Ideas]
XCode 13 introduced Test Repetitions to rerun the same test multiple times based on n number of failures, success, or total count. This would be a great feature to add to test for flakey tests and to make current tests more resilient. 
 
possible example of Scan using test repetitions: 
```
run_tests(
  workspace: "App.xcworkspace",
  scheme: "MyTests",
  clean: false,
  test_repetition: until_success, #until_failure, until_count, false
  max_repetitions: 25,
  repetitions_stop_after_failure: true
)
```

Apple Whitepage: [Diagnose unreliable code with test repetitions](https://developer.apple.com/wwdc21/10296)
Guide on using test repetitions: https://www.avanderlee.com/debugging/flaky-tests-test-repetitions/
Bitrise's implementation: https://blog.bitrise.io/post/xcode-13-test-repetition-options-to-land-on-bitrise

![image](https://user-images.githubusercontent.com/14119319/136091647-abee0da3-a866-4493-914e-2be5fa0aa01b.png)
![image](https://user-images.githubusercontent.com/14119319/136091722-3394924f-451c-415e-8a0c-2ff53725720e.png)

**Comment by osrufung** [comment]
Totally agree scan needs to support this. And the main problem is not the arguments because you can use the test plan, but the output processing. One test with 2 executions, one of them failed will be a failed test even the latest one was successful.

🙏🙏🙏
